### ISSUE TITLE
mroute-api.c: mroute4_del() function does not delete empty wildcard entries

### ISSUE THREAD

**Issue by remedi** [troglobit/smcroute#39]
`mroute4_del()` function does not remove empty wildcard entries from `mroute4_conf_list` and this causes the smcroute logic to continue routing the wildcard entry after the deletion (route matches for new packets). I believe this is not intentional.

```
if (LIST_EMPTY(&mroute4_dyn_list)) {
	entry = LIST_NEXT(entry, link);
	continue;
}
```
Should be something along the lines of:
```
if (LIST_EMPTY(&mroute4_dyn_list)) {
	LIST_REMOVE(entry, link);
	free(entry);
	entry = LIST_FIRST(&mroute4_conf_list);
	continue;
}
```
https://github.com/troglobit/smcroute/blob/master/mroute-api.c#L403

**Comment by troglobit** [comment]
Good catch!  If you could audit commit 50240e6 I can go ahead and merge that fix branch.

**Comment by remedi** [comment]
Kudos for quick fix! In my tests commit 50240e6 fixes the issue.

**Comment by troglobit** [comment]
@remedi No problem, thank you for contributing to free software! :smiley:
